src/circuit/gadget/ecc.rs

```diff
@@ -376,3 +378,168 @@ impl<C: CurveAffine, EccChip: EccInstructions<C> + Clone + Debug + Eq> FixedPoin
+                let base = nullifier_k.0.value();
+                let real_mul_fixed = base * scalar_fixed;
+
+                let scalar_fixed = super::ScalarFixed::new(
+                    chip.clone(),
+                    layouter.namespace(|| "ScalarFixed"),
+                    Some(scalar_fixed),
+                )?;
+                let nullifier_k = super::FixedPoint::get(
+                    chip.clone(),
+                    OrchardFixedBases::NullifierK(nullifier_k),
+                )?;
+                let mul_fixed = nullifier_k.mul(layouter.namespace(|| "mul"), &scalar_fixed)?;
+                if let (Some(x), Some(y)) = (mul_fixed.inner.x.value, mul_fixed.inner.y.value) {
+                    assert_eq!(real_mul_fixed.to_affine(), C::from_xy(x, y).unwrap());
+                }
+            }
+
+            // Check short signed fixed-base scalar multiplication
+            {
+                let scalar_fixed_short = C::Scalar::from_u64(rand::random::<u64>());
+                let mut sign = C::Scalar::one();
+                if rand::random::<bool>() {
+                    sign = -sign;
+                }
+                let scalar_fixed_short = sign * scalar_fixed_short;
+                let value_commit_v = constants::value_commit_v::generator();
+                let real_mul_fixed_short = value_commit_v.0.value() * scalar_fixed_short;
+
+                let scalar_fixed_short = super::ScalarFixedShort::new(
+                    chip.clone(),
+                    layouter.namespace(|| "ScalarFixedShort"),
+                    Some(scalar_fixed_short),
+                )?;
+                let value_commit_v = super::FixedPointShort::get(
+                    chip.clone(),
+                    OrchardFixedBasesShort(value_commit_v),
+                )?;
+                let mul_fixed_short =
+                    value_commit_v.mul(layouter.namespace(|| "mul fixed"), &scalar_fixed_short)?;
+                if let (Some(x), Some(y)) =
+                    (mul_fixed_short.inner.x.value, mul_fixed_short.inner.y.value)
+                {
+                    assert_eq!(real_mul_fixed_short.to_affine(), C::from_xy(x, y).unwrap());
+                }
+            }
+
+            // Check variable-base scalar multiplication
+            {
+                let scalar_val = C::Scalar::rand();
+                let real_mul = p_val * scalar_val;
+
+                let scalar_val = C::Base::from_bytes(&scalar_val.to_bytes()).unwrap();
+                let scalar = super::ScalarVar::new(
+                    chip,
+                    layouter.namespace(|| "ScalarVar"),
+                    Some(scalar_val),
+                )?;
+                let mul = p.mul(layouter.namespace(|| "mul"), &scalar)?;
+                if let (Some(x), Some(y)) = (mul.inner.x.value, mul.inner.y.value) {
+                    assert_eq!(real_mul.to_affine(), C::from_xy(x, y).unwrap());
+                }
+            }
```

Comment:
Check multiplication by 0, and multiplication of a base 𝒪. (I don't think these can occur in the Orchard circuit, but they should either work or have well-defined error behaviour.)

Reply:
`[0]B` returns `(0,0)` since variable-base scalar mul uses complete addition for the final bits. `[a]𝒪` returns an error since it uses incomplete addition for the initial bits.

Reply:
That sounds fine. The latter should be documented.